### PR TITLE
Refreshing Cockatrice Sound

### PR DESCRIPTION
**Pull request by poixen**
+ added sound setting page
+ added sound setting icon
+ moved sound settings from interface settings
+ added master volume slider and spin box
+ audio preview when dragging and releasing the slider
+ **removed all old sfx**
+ added new end step and tap sfx
+ tap/endstep sound has a timer on it to prevent spamming
+ test sound engine will now use endstep sfx

sounds2

### CODE REVIEW

**Comment by poixen** [comment]
Added master volume slider for global app control
Can be expanded upon to only changed certain sounds.

**Comment by tooomm** [comment]
CUCKOO!

**Comment by poixen** [comment]
Why does qt4 suck so much, you cant set the volume?

File: cockatrice/src/dlg_settings.cpp [hunk 1]
```diff
@@ -718,6 +676,102 @@ void MessagesSettingsPage::retranslateUi()
...
+    soundPathEdit->setText(QString());
+    settingsCache->setSoundPath(QString());
+}
+
+void SoundSettingsPage::soundPathButtonClicked()
+{
+    QString path = QFileDialog::getExistingDirectory(this, tr("Choose path"));
+    if (path.isEmpty())
+        return;
+
```
Comment (Daenyth) [inline cockatrice/src/dlg_settings.cpp]: We might want to generate some feedback here

Comment (poixen) [inline cockatrice/src/dlg_settings.cpp]: Sure, that was just a direct copypasta from the other page

**Comment by poixen** [comment]
This is ready to be tested, Im quite happy with it.

**Comment by tooomm** [comment]
is the new player sound also played for spectators joining?

**Comment by poixen** [comment]
Yes.

**Comment by ZeldaZach** [comment]
Just a personal suggestion; The volume icon at the top should be flipped so it's outward like this 

test

**Comment by tooomm** [comment]
no more shuffle / draw / notification / play card sounds? ;(

just join / tap+untap / end step?

**Comment by poixen** [comment]
Correct. This feels the most comfortable to use. If there is enough demand for them, I might put something in. Right now, this feels good.

**Comment by LestatTheDarkPrince** [comment]
Would like to voice a vote for the original sounds to be fixed.  Killing 90% of the sounds which were not a problem seems very dumb. the shuffle, draw, notification, and play sounds need to come back. I may comment later once i see what you did seeing as i am forced to play soundless for 1 week.

**Comment by poixen** [comment]
Sorry you don't like the changes. You can revert back to an older build if you would like the old sounds. This was a total overhaul of sounds and when sounds play. Again, older versions are available.

**Comment by Daenyth** [comment]
Eventually we will get a sound theme system going and you can use the
original sounds if you like. I can't say when

**Comment by LestatTheDarkPrince** [comment]
I have an older version sound file, it does not seem to work, the best I got are the 3 new sounds which are LOW even on full blast they seem low.The games are just too quiet now.

**Comment by tooomm** [comment]
Being able to enable/disable single sounds would be awesome!!
We have the Sound tab now in settings, plenty of space there for some check boxes. ;P


<br>
I like to see some of the old sounds in again then.
I miss the sound for desktop notifications for example. Just a tiny & comfortable sound to get your attention like some smartphone apps, desktop messengers or gitter use would be great here. No big bling bling! ;)

Shuffle and especially draw was nice too. Since it was kinda easy to notice without reading the log if your opponent happened to draw two cards by fault. Happens sometimes if you use your mouse to click the phases:
- click draw --> switch to phase
- click draw --> draw a card
- click draw --> draw a card
- ...<br><br>
If you play and click fast you sometimes happen to click too often or when there is a server delay for example.

**Comment by LestatTheDarkPrince** [comment]
I agree with all of your points. The sound tab has plenty of space as you
said. If they do the check boxes for what sound to hear then its a small
step to have a drop down box where you can set the sound(s) from various
sounds on your machine. You can make them whatever. And also set them at
volumes for each. i am a rules Natzie so i am watching for people who cheat
if by accident or on purpose hoping I would not catch them. I also use my
mouse for everything.

**Pull request merged**